controllers/taloscontrolplane_controller.go

```diff
@@ -747,19 +747,27 @@ func (r *TalosControlPlaneReconciler) updateStatus(ctx context.Context, tcp *con
 		return nil
 	}
 
-	kubeclient, err := r.kubeconfigForCluster(ctx, util.ObjectKey(cluster))
-	if err != nil {
-		return err
-	}
-
-	defer kubeclient.Close() //nolint:errcheck
-
 	conditionGetters := make([]conditions.Getter, len(ownedMachines))
 
 	for i, v := range ownedMachines {
 		conditionGetters[i] = &v
 	}
 
+	defer func() {
+		conditions.SetAggregate(tcp, controlplanev1.MachinesReadyCondition, conditionGetters, conditions.AddSourceRef(), conditions.WithStepCounterIf(false))
+	}()
+
+	kubeclient, err := r.kubeconfigForCluster(ctx, util.ObjectKey(cluster))
+	if err != nil {
+		r.Log.Error(err, "failed to get kubeconfig for the cluster")
+
+		tcp.Status.UnavailableReplicas = replicas
+
+		return nil
```

Comment:
how will controller know that it's time to run another reconcile loop when `kubeconfig` is available?

I always thought that core CAPI creates the `kubeconfig` based on the CA secret which is created by the bootstrap controller, so it should be more or less always available?

Reply:
well, apparently it's only available much later.

Somehow it's able to reconcile it in the right way.